### PR TITLE
Feature add lock scale option

### PR DESCRIPTION
**Pull request by Zyrck56**
Feature implemented and working to the best of my knowledge.
Changes also include a typo fix, a 33ms wait time instead of 30ms, and magic number removal.

### CODE REVIEW

File: FESS-GUI/mainwindow.h [hunk 1]
```diff
@@ -31,6 +31,7 @@ class MainWindow : public QMainWindow
     QTimer *accelerationSlopeTimer;
     bool playSounds = false;
     bool isRecording = false;
+    bool isScaleLocked = false;
```
Comment (ghost) [inline FESS-GUI/mainwindow.h]: Do we want to initialize variables in the headers or do it in the constructor?

**Review by ghost — COMMENTED** [review]
Code looks good other than my question about headers.

File: FESS-GUI/graph.cpp [hunk 2]
```diff
@@ -84,15 +85,21 @@ void ScrollingTimeGraph::addData(QCustomPlot* plot, double time, double primaryD
...
     // remove data of lines that's outside visible range:
-    plot->graph(0)->removeDataBefore(time-8);
-    plot->graph(1)->removeDataBefore(time-8);
+    plot->graph(0)->removeDataBefore(time-range);
+    plot->graph(1)->removeDataBefore(time-range);
 
     // rescale value (vertical) axis to fit the current data:
-    plot->graph(0)->rescaleValueAxis();
-    plot->graph(1)->rescaleValueAxis(true);
+    if(maxValue < 0){
```
Comment (JCJutson) [inline FESS-GUI/graph.cpp:92]: Currently, it is possible that maxValue will have a negative value. We should look into fixing that. There are some weird consequences. I'll bring it up in the meeting

File: FESS-GUI/mainwindow.h (hunk 1 above)
Comment (JCJutson) [reply FESS-GUI/mainwindow.h]: Yeah, let's be consistent with how we do this

File: FESS-GUI/mainwindow.cpp [hunk 3]
```diff
@@ -152,8 +152,14 @@ void MainWindow::realtimeDataSlot()  //Important function. This is repeatedly ca
     ui->label_12->setText(selectedGraph->currentDisplay());
 
     //add data to graphs
-    velocityGraph->addData(currentTime, actualVelocity, radsPerSecondToRPM(currentExpectedVelocity));
-    accelerationGraph->addData(currentTime, actualAcceleration, currentExpectedAcceleration);
+    if(isScaleLocked){
+        velocityGraph->addData(currentTime, actualVelocity, radsPerSecondToRPM(currentExpectedVelocity), ui->velocitySlider->maximum());
```
Comment (JCJutson) [inline FESS-GUI/mainwindow.cpp]: I think we discussed using the slider maximum + 5% for the graph maximum

**Review by JCJutson — CHANGES_REQUESTED** [review]

**Review by JCJutson — APPROVED** [review]

File: FESS-GUI/mainwindow.cpp [hunk 4]
```diff
@@ -557,7 +562,7 @@ void MainWindow::on_actionLock_frame_rate_at_30FPS_triggered(bool checked)
 {
     if(checked)
     {
-        graphRefreshRate = 30;
+        graphRefreshRate = 33;
```
Comment (ghost) [inline FESS-GUI/mainwindow.cpp]: Refresh Rates are now saved as FPS/HZ. The original 30 would be correct. 1000/30 is done in a function to get the 33ms for the timer.

**Review by ghost — CHANGES_REQUESTED** [review]

File: FESS-GUI/mainwindow.cpp (hunk 4 above)
Comment (Zyrck56) [reply FESS-GUI/mainwindow.cpp]: okay. fixed

**Review by ghost — APPROVED** [review]

**Pull request merged**